pkgs/development/tools/rust/cargo-zigbuild/default.nix

```
@@ -0,0 +1,29 @@
+{ lib, rustPlatform, fetchFromGitHub, makeWrapper, zig }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "cargo-zigbuild";
+  version = "0.12.0";
+
+  src = fetchFromGitHub {
+    owner = "messense";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-nBncU5rM3gS5e/Qs14U/ZwAkLFLdNuO2DhSQW+7xGQk=";
+  };
+
+  cargoSha256 = "sha256-Zq+RG36aeNd8G+LSdiyLK8SYC0MckGUIBTvia4H9OJY=";
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  postInstall = ''
+    wrapProgram $out/bin/cargo-zigbuild \
+      --prefix PATH : ${zig}/bin
+  '';
+
+  meta = with lib; {
+    description = "A tool to compile Cargo project with zig as linker";
+    homepage = "https://github.com/messense/cargo-zigbuild";
+    license = licenses.mit;
+    maintainers = with maintainers; [ figsoda ];
```

Pretty sure this can also build on Darwin.
```suggestion
    maintainers = with maintainers; [ figsoda ];
    platforms = platforms.unix;
```

---

this is set by `buildRustPackage`: https://github.com/NixOS/nixpkgs/blob/2da64a81275b68fdad38af669afeda43d401e94b/pkgs/build-support/rust/build-rust-package/default.nix#L157